app/models/student.rb

```diff
@@ -1,16 +1,17 @@
 class Student < ActiveRecord::Base
-  attr_accessible :title, :birth_date, :email, :first_name, :gender, :last_name, :middle_name
+  attr_accessible :title_id, :birth_date, :email, :first_name, :gender, :last_name, :middle_name
 
   belongs_to :title
 
+  validates :title, presence: true
   validates :first_name, presence: true
   validates :last_name, presence: true
   validates :email, presence: true, uniqueness: true
   validates :birth_date, presence: true
   validates :gender, presence: true
 
   def full_name
-    [title, first_name, middle_name, last_name].join(' ')
+    [title.name, first_name, middle_name, last_name].join(' ')
```

Comment:
There exists the edge case where a student might not have a title, therefor when accessing the name attribute you would get an error e.g.
```
NoMethodError: undefined method 'name' for nil:NilClass
```
To get around this you can use the safe navigation operator, e.g.
```
[title&.name, first_name, middle_name, last_name].join(' ')
```
Lastly, in the Jobready Plus project you'll typically see such view methods like `full_name` in a decorator class. Check out the `draper` gem on github for more information on how to do that, but in this case you'd have another class like below:
```
class StudentDecorator < Draper::Decorator
  delegate_all

  def full_name
    [object.title&.name, object.first_name, object.last_name].join(' ')
  end
end

Reply:
Thanks. Will make these changes 👍